src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.BufferPoolEventSource",
+        LocalizationResources = "FxResources.System.Buffers.SR")]
+    internal sealed class ArrayPoolEventSource : EventSource
+    {
+        internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
+
+        private const EventTask Buffer = (EventTask)1;
+
+        [Event(1, Level = EventLevel.Informational, Opcode = EventOpcode.Start, Task = Buffer)]
+        internal void BufferRented(int bufferId, int bufferSize, int poolId, int bucketId) { WriteEvent(1, bufferId, bufferSize, poolId, bucketId); }
```

Comment:
You can double check, but I don't believe there's a WriteEvent(int, int, int, int, int) overload, which means this is actually binding to the WriteEvent(int, params object[]) overload.  That's going to add boxing and array allocations when tracing is enabled.  Consider providing a more efficient implementation... you can see a very similar example here: https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/Threading/Tasks/TPLETWProvider.cs#L228-L251